### ISSUE TITLE
It performed very well on the trainset, but performed very poorly on the val set

### ISSUE THREAD

**Issue by YXLH7** [zimyang/MaskRelation#2]
Has anyone successfully run this paper? I ran the pytorch version and it performed very well on the training set, but it performed very poorly on the validation set, can anyone help me?

**Comment by wym-dlone** [comment]
Hello, is it convenient to share the code of data preprocessing and the structure of data storage? Thank you very much

**Comment by zimyang** [comment]
> Has anyone successfully run this paper? I ran the pytorch version and it performed very well on the training set, but it performed very poorly on the validation set, can anyone help me?

Could you describe the problem in detail? The model is possibly overfitted. Is the model trained on the whole training dataset?

**Comment by zimyang** [comment]
> Hello, is it convenient to share the code of data preprocessing and the structure of data storage? Thank you very much

Code of data preprocessing is [detect_from_video.py](https://github.com/ondyari/FaceForensics/blob/master/classification/detect_from_video.py).
The structure of data storage:
data.json
--video name 1
----list (data type: List. A list of filenames of frames)
----label (data type: int. True is 0. False is 1)
--video name 2

**Comment by umar-umich** [comment]
@zimyang thanks for the response but [detect_from_video.py ](https://github.com/ondyari/FaceForensics/blob/master/classification/detect_from_video.py)  don't create data.json... Can you please provide the modified preprocessing code tailored for your repo? Thanks

**Comment by liruotong123** [comment]
@YXLH7 Have you already run through this code and can you give me a brief description of the training process?

**Comment by AdaxKK** [comment]
@devloverumar Hello，Do you have the relevant json file？

**Comment by zimyang** [comment]
How to get these data.json?
**1. Extract facial images。** Refer to [extract_images.py](https://github.com/ondyari/FaceForensics/tree/original/extract_images.py) 和 [detect_from_video.py](https://github.com/ondyari/FaceForensics/tree/master/classification/detect_from_video.py).
(1) Read videos of the dataset. 
(2) Extract the facial regions from video frames by face detectors like Dlib.
(3) Save the facial regions as image files.

**2. Create data.json**
(1) Read paths of the images.
(2) Get the ground truth of videos from the database.
The format of data.json:
[
{
key: "ffpp001",
value: {'list': ["ffpp001/ffpp001_001.jpg", "ffpp001/ffpp001_002.jpg", "ffpp001/ffpp001_003.jpg"],
'label': 0}
},
{
key: "ffpp002",
value: {'list': ["ffpp002/ffpp002_001.jpg", "ffpp002/ffpp002_002.jpg", "ffpp002/ffpp002_003.jpg"],
'label': 1}
},
...]

By the way, my email has been expired because of graduation. Welcome to contact me via yangziming20[at]mails[dot]ucas.ac.cn
Thanks for your support!
